File: Magic Number/file.c
#include <stdio.h>

int magic(int n){
    int x=n,sum=0,rev=0;
    
    //getting sum of digits
    while(x>0){
        sum+=x%10;
        x/=10;
    }
    
    //getting reversed value for the sum of digits
    int sum1=sum;
    while(sum1>0){
        rev=rev*10+sum1%10;
        sum1/=10;
    }
    return rev*sum;
}
int main()
{
    int num;
    scanf("%d",&num);
    int x=(magic(num)==num);
    if(x==1){
        printf("Yes it's a Magic Number");
    }
    else{
        printf("No it's not a Magic Number");
    }

    return 0;
}

File: Magic Number/file.py
def magic(n):
    x,sumval,rev=n,0,0
    
    #getting sum of digits
    while(x>0):
        sumval=sumval+x%10
        x=int(x/10)
        
    #getting reverse value for the sum of digits
    sumval1=sumval
    while(sumval1>0):
        rev=rev*10+sumval1%10
        sumval1=int(sumval1/10)
    
    return sumval*rev

num=int(input())
x=(magic(num)==num)
if(x==1):
    print("Yes it's a Magic Number")
else:
    print("No it's not a Magic Number")
